services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/ProtectedPathSpec.groovy

```diff
@@ -668,6 +693,41 @@ class ProtectedPathSpec extends BaseSpecification {
 
         currentLastUpdate < northbound.getFlow(flow.id).lastUpdated
 
+        and: "New meter is created on the src and dst switches"
+        def newSrcSwitchCreatedMeterIds = getCreatedMeterIds(srcSwitch.dpId)
+        def newDstSwitchCreatedMeterIds = getCreatedMeterIds(dstSwitch.dpId)
+        newSrcSwitchCreatedMeterIds != srcSwitchCreatedMeterIds
+        newDstSwitchCreatedMeterIds != dstSwitchCreatedMeterIds
+
+        and: "Rules are updated"
+        Wrappers.wait(WAIT_OFFSET) { flowHelper.verifyRulesOnProtectedFlow(flow.id) }
+
+        and: "Old meter is deleted on the src and dst switches"
+        Wrappers.wait(WAIT_OFFSET) {
+            [srcSwitch.dpId, dstSwitch.dpId].each { switchId ->
+                def switchValidateInfo = northbound.switchValidate(switchId)
+                assert switchValidateInfo.meters.proper.size() == 1
+                assert switchValidateInfo.rules.proper.size() == 3
+                switchHelper.verifyRuleSectionsAreEmpty(switchValidateInfo, ["missing", "excess"])
+                switchHelper.verifyMeterSectionsAreEmpty(switchValidateInfo, ["missing", "misconfigured", "excess"])
+            }
+        }
+
+        and: "Transit switches store the correct info about rules and meters"
+        def involvedTransitSwitches = (currentPath[1..-2].switchId + currentProtectedPath[1..-2].switchId).unique()
+        Wrappers.wait(WAIT_OFFSET) {
+            involvedTransitSwitches.each { switchId ->
+                def amountOfRules = (switchId in currentProtectedPath*.switchId && switchId in currentPath*.switchId)
+                        ? 4 : 2
+                def switchValidateInfo = northbound.switchValidate(switchId)
+                assert switchValidateInfo.rules.proper.size() == amountOfRules
+                switchHelper.verifyRuleSectionsAreEmpty(switchValidateInfo, ["missing", "excess"])
+                switchHelper.verifyMeterSectionsAreEmpty(switchValidateInfo)
+            }
+        } || true
```

Comment:
`|| true` can be dropped since it is applied to `wait`

Reply:
I'll move it on a line above

Reply:
No need, since there are no implicit assertions inside `wait`. you can just drop it

Reply:
fixed